libkbfs/folder_block_ops.go

```diff
@@ -839,19 +847,19 @@ func (fbo *folderBlockOps) fixChildBlocksAfterRecoverableError(
 
 		fbo.log.CDebugf(ctx, "Re-dirtying %v (and deleting dirty block %v)",
 			newPtr, oldPtr)
-		// These block would have been permanent,
-		// so they're definitely still in the
-		// cache
-		b, err := bcache.Get(newPtr, fbo.branch())
+		// These block would have been permanent, so they're
+		// definitely still in the cache
+		b, err := fbo.getBlockFromCache(newPtr, fbo.branch())
 		if err != nil {
 			fbo.log.CWarningf(ctx, "Couldn't re-dirty %v: %v", newPtr, err)
 			continue
 		}
-		err = bcache.PutDirty(newPtr, fbo.branch(), b)
+		dbcache := fbo.config.DirtyBlockCache()
+		err = dbcache.Put(newPtr, fbo.branch(), b)
```

Comment:
There's a few `Put` calls into the dirty cache, but we still leave the blocks in the clean cache. Is this correct behavior?